lib/miq_expression.rb

```diff
@@ -1554,36 +1554,43 @@ def self._quick_search?(e)
     end
   end
 
-  def custom_attribute_columns_and_models(expression = nil)
-    return custom_attribute_columns_and_models(exp).uniq if expression.nil?
+  def custom_attribute_columns_and_models
+    @virtual_custom_attributes ||= begin
+      @virtual_custom_attributes = []
+      _custom_attribute_columns_and_models
+      @virtual_custom_attributes
+    end
+  end
+
+  def _custom_attribute_columns_and_models(expression = nil)
+    return _custom_attribute_columns_and_models(exp) if expression.nil?
 
     case expression
     when Array
-      expression.flat_map { |x| custom_attribute_columns_and_models(x) }
+      expression.flat_map { |x| _custom_attribute_columns_and_models(x) }
     when Hash
       expression_values = expression.values
-      return [] unless expression.keys.first
-
-      if expression.keys.first == "field"
-        begin
-          field = Field.parse(expression_values.first)
-        rescue StandardError => err
-          _log.error("Cannot parse field #{field}" + err.message)
-          _log.log_backtrace(err)
+      return unless expression.keys.first
+
+      if expression.keys.first == "field" || expression.keys.second == "value"
+        left_field, right_field = expression_values.map do |x|
+          begin
+            Field.parse(x) if x.to_s.include?(CustomAttributeMixin::CUSTOM_ATTRIBUTES_PREFIX)
+          rescue StandardError => err
```

Comment:
changed to new valid_custom_attribute?
yea but difference is now that on right side can be value so basically it can be everything not just column in our format "Model-attribute", and Field.parse was causing exception here so it is a reason I added check for the prefix.